htd/utils.py

```diff
@@ -0,0 +1,164 @@
+from htd.constants import HtdConstants
+from htd.models import ZoneDetail
+
+
+# the command sequence we use to send to the device, the header and reserved
+# bytes are always first, the zone is third, followed by the command and code.
+def get_command(zone: int, command: bytes, data_code: int) -> bytes:
+    cmd = [
+        HtdConstants.HEADER_BYTE,
+        HtdConstants.RESERVED_BYTE,
+        zone,
+        command,
+        data_code,
+    ]
+    checksum = calculate_checksum(cmd)
+    cmd.append(checksum)
+
+    return bytes(cmd)
+
+
+# convert the volume into a usable value. the device will transmit a number
+# between 196 - 255. if it's at max volume, the raw volume will come as 0.
+# this is probably because the gateway only transmits 8 bits per byte.
+# 255 is 0b11111111. since there's no volume = 0 (use mute I guess), if the
+# volume hits 0, it's because it's at max volume, so we make it 256.
+# credit for this goes to lounsbrough
+def convert_volume(raw_volume: int) -> (int, int):
```

Comment:
**nitpick:** imo, in-line comments are better with this simple formatting. I would use doc strings for functions so that you can parse the info if you ever care to.
```suggestion
def convert_volume(raw_volume: int) -> (int, int):
    """
    convert the volume into a usable value. the device will transmit a number
    between 196 - 255. if it's at max volume, the raw volume will come as 0.
    this is probably because the gateway only transmits 8 bits per byte.
    255 is 0b11111111. since there's no volume = 0 (use mute I guess), if the
    volume hits 0, it's because it's at max volume, so we make it 256.
    credit for this goes to lounsbrough
    """
```